iceoryx2/src/service/static_config/message_type_details.rs

```diff
@@ -120,3 +122,124 @@ impl MessageTypeDetails {
             && self.payload.alignment <= rhs.payload.alignment
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use iceoryx2_bb_testing::assert_that;
+
+    #[test]
+    fn test_from() {
+        struct MyPayload {
+            _a: i32,
+            _b: bool,
+            _c: i64,
+        }
+
+        let got = MessageTypeDetails::from::<i32, i64, MyPayload>(TypeVariant::FixedSize);
+        let want = MessageTypeDetails{
+            header:  TypeDetail{
+                variant: TypeVariant::FixedSize,
+                type_name: "i32".to_string(),
+                size: 4,
+                alignment: 4,
+            },
+            user_header: TypeDetail{
+                variant: TypeVariant::FixedSize,
+                type_name: "i64".to_string(),
+                size: 8,
+                alignment: 8,
+            },
+            payload: TypeDetail{
+                variant: TypeVariant::FixedSize,
+                type_name: "iceoryx2::service::static_config::message_type_details::tests::test_from::MyPayload".to_string(),
+                size: 16,
+                alignment: 8,
+            },
+        };
+        assert_that!(want, eq got);
+
+        let got = MessageTypeDetails::from::<i32, bool, String>(TypeVariant::Dynamic);
+        let want = MessageTypeDetails {
+            header: TypeDetail {
+                variant: TypeVariant::FixedSize,
+                type_name: "i32".to_string(),
+                size: 4,
+                alignment: 4,
+            },
+            user_header: TypeDetail {
+                variant: TypeVariant::FixedSize,
+                type_name: "bool".to_string(),
+                size: 1,
+                alignment: 1,
+            },
+            payload: TypeDetail {
+                variant: TypeVariant::Dynamic,
+                type_name: "alloc::string::String".to_string(),
+                size: 24,
+                alignment: 8,
+            },
+        };
+        assert_that!(want, eq got);
+    }
+
+    #[test]
+    fn test_user_header_ptr_from_header() {
+        let details = MessageTypeDetails::from::<i32, bool, String>(TypeVariant::Dynamic);
+        struct Demo {
```

Comment:
Please add `#[repr(C)]`. Here it is again important, otherwise the compiler may reorder the member variables and then the pointer calculations are wrong.